enhancements/machine-config/mco-flattened-ignition.md

```diff
@@ -0,0 +1,178 @@
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposes adding an interface to the MCO such that a flattened
+ignition config, that is a combination of the managed pointer config and the
+MCO rendered config (inlined via a [data URL](https://tools.ietf.org/html/rfc2397))
+is available.
+
+## Motivation
+
+In some situations it is necessary to perform network configuration before it
+is possible to download the rendered config from the MCS in the ramdisk where
+ignition evaluates the pointer config.
+
+This leads to a chicken-egg problem, where a user could configure the network
+via ignition, but ignition cannot apply any config until it has downloaded
+any append/merge directives.
+
+This situation is quite common in baremetal environments, e.g a bonded pair of nics
+(where the native vlan is often used for PXE provisioning),
+with the controlplane network configured via a VLAN on the same physical interface.
+
+For UPI deployments, it may be possible to work around this situation when
+using coreos-install following the [enhancements around static networking](https://github.com/openshift/enhancements/blob/master/enhancements/rhcos/static-networking-enhancements.md), but it may be a more consistent user
+experience if all configuration was possible via ignition.
+
+For baremetal IPI deployments currently the only workaround is to customize the
+OS image, since the deploy toolchain currently does not use coreos-install,
+or include any interface to drop additional files into /boot during deployment.
+
+### Goals
+
+ * Provide a solution to customer requirements for baremetal IPI around bond/VLAN
+   and other configurations for controlplane networking
+ * Avoid a platform-specific approach, which could be fragile and/or hard to maintain
+
+### Non-Goals
+
+The discussion of interfaces here only considers the existing interfaces, not
+proposals around a future [declarative network configuration](https://github.com/openshift/enhancements/pull/399)
+although we need to ensure the approach taken in each case is aligned.
+
+### User Stories
+
+#### Story 1
+
+As a UPI user, I would like to perform network configuration in the same
+way as other customization, e.g via MachineConfig manifests or ignition.
+
+#### Story 2
+
+As a baremetal IPI user, I need to deploy in an environment where network
+configuration is required before access to the controlplane network is possible.
+
+## Design Details
+
+The MCO currently maintains two ignition artifacts, a managed instance of the
+pointer ignition config, stored in a Secret, and the rendered config stored
+in a MachineConfig object (both per role, not per machine).
+
+Both artifacts are accessible via the MCS (although not currently from within
```

Comment:
uhm, the pointer ignition config isn't "accessible" via the MCS so s/both artifacts/just the rendered config/g